week2/length.c

```diff
@@ -0,0 +1,80 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdbool.h>
+#include "length.h"
+
+//Function Definitions
+Input GetUserInput(char* input)
+{
+    //initialize struct Input 
+    Input userInput;
+    
+    userInput.stringLength = strlen(input);
+    
+    char* actualString = malloc(sizeof(*actualString) * userInput.stringLength);
+    strcpy(actualString, input);
+    userInput.inputString = actualString;
+    
+    //Check the values being put in struct Input
+    printf("string length = %d\n", userInput.stringLength);
+    printf("input string = %s\n", userInput.inputString);
+    
+    return(userInput);
+}
+
+int LongestSubstringNoRepeatChar(Input stringInfo)
+{
+    int left = 0; //left pointer in an array of characters
+    int right = 1; //right pointer in an array of characters
+    int length = stringInfo.stringLength; 
+    char* inputString = stringInfo.inputString;
+    
+    //The input string is not valid
+    if(inputString == NULL)
+    {
+    	return(-1); 
+	}
+    
+    //No need to check when there is only 1 character
+    if(length == 1)
+    {
+    	return(1);
+	}
+	
+	//when the string has more than 1 character
+	int longestLength = 1; //set to 1 first, the minimum length
+	int currentLength = 1;
+	while (right < length)
+    {
+   		char* charAddress = strchr(&(inputString[left]), inputString[right]);
```

Comment:
Think a bit more about the time complexity here - what would "strchr" be doing under the hood?